src/securityPolicy.c

```diff
@@ -634,51 +636,81 @@ security_policy_t policyForSignTxStakePoolRegistrationConfirm(
 }
 
 // For each withdrawal
-security_policy_t policyForSignTxWithdrawal()
+security_policy_t policyForSignTxWithdrawal(
+        sign_tx_usecase_t signTxUsecase,
+        const stake_credential_type_t stakeCredentialType,
+		const bip44_path_t* stakingKeyPath
+)
 {
-	// No need to check withdrawals
-	SHOW();
+	switch (signTxUsecase) {
+	case SIGN_TX_USECASE_ORDINARY_TX:
+		DENY_UNLESS(stakeCredentialType == STAKE_CREDENTIAL_KEY_PATH);
+		ASSERT(stakingKeyPath != NULL);
+		DENY_UNLESS(bip44_isOrdinaryStakingKeyPath(stakingKeyPath));
+		ALLOW();
+		break;
+
+	case SIGN_TX_USECASE_MULTISIG:
+		DENY_UNLESS(stakeCredentialType == STAKE_CREDENTIAL_SCRIPT_HASH);
+		ASSERT(stakingKeyPath == NULL);
+		ALLOW();
+		break;
+
+	default:
+		ASSERT(false);
+	}
+
+	DENY(); // should not be reached
 }
 
 
+static inline security_policy_t nonPoolSubPolicy(const bip44_path_t* pathSpec)
```

Comment:
It's called from the transaction use cases that are not pool registration. It's just a static function, so I think context can be used instead of an overly verbose name, but if you have a good suggestion, I'm all for it.

Reply:
```
	case SIGN_TX_USECASE_ORDINARY_TX: {

		switch (bip44_classifyPath(pathSpec)) {

		case PATH_WALLET_SPENDING_KEY:
		case PATH_WALLET_STAKING_KEY:
		case PATH_POOL_COLD_KEY:
			if (bip44_isPathReasonable(pathSpec)) {
				ALLOW();
			} else {
				WARN();
			}
			break;

		default:
			DENY();
			break;
		}

		break;
	}

```
This is how it looked before. The nonpool refers to the tx usecase, not the path's poolness.

Reply:
This `nonPoolSubPolicy` is no longer needed?